main.rb

```diff
@@ -0,0 +1,152 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    array = to_a
+    index = 0
+    while index < size
+      yield(array[index])
+      index += 1
+    end
+    array
+  end
+
+  #--------my each by index -------
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    array = to_a
+    index = 0
+    while index < size
+      yield(array[index], index)
+      index += 1
+    end
+    array
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    select = []
+    my_each { |i| select.push(i) if yield(i) }
+    select
+  end
+
+  def my_all?(args = nil)
+    if !args.nil? # if arguments or method parameters were given
+      case args
+      when Class
+        my_each { |i| return false unless i.is_a?(args) }
+      when Regexp
+        my_each { |i| return false unless args.match(i.to_s) }
+      else
+        my_each { |i| return false unless i == args }
+      end
+    elsif block_given? # if block was given
+      my_each { |i| false unless yield(i) }
+    else
+      my_each { |i| false unless i }
+    end
+    true
+  end
+
+  def my_any?(args = nil)
+    if !args.nil? # if arguments or method parameters were given
+      case args
+      when Class
+        my_each { |i| return true if i.is_a?(args) }
+      when Regexp
+        my_each { |i| return true if args.match(i.to_s) }
+      else
+        my_each { |i| return true if i == args }
+      end
+    elsif block_given? # if block was given
+      my_each { |i| true if yield(i) }
+    else
+      my_each { |i| true if i }
+    end
+    false
+  end
```

Comment:
There is a problem with this method. It should  return true if the block ever returns a value other than false or nil. Kindly double check on this